runtime/runtime.h

```diff
@@ -169,4 +173,80 @@ void stub_init();
 // The runtime entrypoint must be called
 int runtime_main(int argc, char** argv);
 
+#ifdef DYNAMIC_LINKING_WASM_SO
+
+typedef void (*awsm_abi__init_globals_fn_t)(void);
+typedef void (*awsm_abi__init_mem_fn_t)(void);
+typedef void (*awsm_abi__init_tbl_fn_t)(void);
+typedef int32_t (*awsm_abi__entrypoint_fn_t)(void);
+typedef uint32_t (*awsm_abi__wasm_memory_starting_pages_fn_t)(void);
+typedef uint32_t (*awsm_abi__wasm_memory_max_pages_fn_t)(void);
+typedef uint32_t (*awsm_abi__wasm_memory_globals_len_fn_t)(void);
+
+struct awsm_abi_symbols {
```

Comment:
I can't quite fit it in my head why this is the abi, and other things are not. I think I'm just lacking context here.

Reply:
changed